spec/support/resource_behavior.rb

```diff
@@ -0,0 +1,51 @@
+require 'blockscore/util'
+
+RSpec.shared_context 'a resource' do
+  let(:resource) do
+    BlockScore::Util.to_underscore(described_class.to_s[/BlockScore::(\w+)/, 1])
+  end
+
+  it 'test_create_resource' do
+    create(:"#{resource}_params")
+    assert_requested(@api_stub, times: 1)
+  end
+
+  it 'test_retrieve_resource' do
+    r = build(resource)
+
+    expect(resource).to eq(described_class.retrieve(r.id).object)
+    assert_requested(@api_stub, times: 1)
+  end
+
+  it 'test_list_resource' do
+    described_class.all
+
+    assert_requested(@api_stub, times: 1)
+  end
+
+  it 'test_list_resource_with_count' do
+    response = described_class.all(count: 2)
+
+    expect(2).to eq(response.count)
+    assert_requested(@api_stub, times: 1)
+  end
+
+  it 'test_list_resource_with_count_and_offset' do
+    response = described_class.all(count: 2, offset: 2)
+
+    expect(2).to eq(response.count)
+    assert_requested(@api_stub, times: 1)
+  end
+
+  it 'test_init_and_save' do
+    params = build(resource).attributes
+    obj = described_class.new
+
+    params.each do |key, value|
+      obj.public_send "#{key.to_s}=".to_sym, value
```

Comment:
Lint/StringConversionInInterpolation: Redundant use of `Object#to_s` in interpolation.